### PR TITLE
docs #747 reconciliation

### PR DESCRIPTION
**Pull request by oraNod**
Docs for #747

### CODE REVIEW

**Review by dvagnero — APPROVED** [review]
Nice improvement

File: documentation/asciidoc/topics/ref_datagrid_service_resources.adoc [hunk 1]
```diff
@@ -53,6 +53,13 @@ include::yaml/datagrid_service.yaml[]
 |`spec.configMapName`
 |Specifies a `ConfigMap` that contains {brandname} configuration.
 
+|`spec.configListener`
+|Creates a `listener` pod in each {brandname} cluster that allows {ispn_operator} to reconcile server-side modifications with `Infinispan` resources such as the `Cache` CR.
+
+The `listener` pod consumes minimal resources and is enabled by default.
+Setting a value of `false` removes the `listener` pod and disables reconciliation.
+You should do this only if you access `Infinispan` resources through the {brandname} Console, CLI, or client applications.
```
Comment (ryanemerson) [inline documentation/asciidoc/topics/ref_datagrid_service_resources.adoc:61]: I think this can be made a little clearer, how about:

"You should do this only if you do not require declarative Kubernetes representations of `Infinispan` resources created via the {brandname} Console, CLI, or client applications."

Also, should `Infinispan` be parametised for downstream?

File: documentation/asciidoc/topics/con_operator_deployments.adoc [hunk 2]
```diff
@@ -29,3 +31,12 @@ endif::community[]
...
 endif::downstream[]
+
+= Resource reconciliation
+
+{ispn_operator} reconciles `Infinispan` custom resources such as the `Cache` CR with resources on your {brandname} cluster.
+
+Bidirectional reconciliation synchronizes your CRs with changes that you make to `Infinispan` resources through the {brandname} Console, command line interface (CLI), or other client application and vice versa.
+For example if you create a cache through the {brandname} Console then {ispn_operator} adds a declarative Kubernetes representation.
+
+To perform reconciliation {ispn_operator} creates a `listener` pod in each {brandname} cluster that detects modifications for `Infinispan` resources.
```
Comment (ryanemerson) [inline documentation/asciidoc/topics/con_operator_deployments.adoc:42]: `/s/in each/for each`

File: documentation/asciidoc/topics/con_operator_deployments.adoc [hunk 3]
```diff
@@ -29,3 +31,12 @@ endif::community[]
 ifdef::downstream[]
 image::operator-managed-clusters.png[This illustration depicts how {ispn_operator} manages multiple clusters on {openshiftshort}.]
 endif::downstream[]
+
+= Resource reconciliation
+
+{ispn_operator} reconciles `Infinispan` custom resources such as the `Cache` CR with resources on your {brandname} cluster.
+
+Bidirectional reconciliation synchronizes your CRs with changes that you make to `Infinispan` resources through the {brandname} Console, command line interface (CLI), or other client application and vice versa.
+For example if you create a cache through the {brandname} Console then {ispn_operator} adds a declarative Kubernetes representation.
```
Comment (ryanemerson) [inline documentation/asciidoc/topics/con_operator_deployments.adoc:40]: It's probably worth noting somewhere that Cache CRs created via the ConfigListener have their owner reference set to the Infinispan CR which they are associated with, meaning that when the Infinispan CR is deleted the cache CR(s) will be as well. Conversely, Cache CRs created by the user are not deleted with the Infinispan CR.

File: documentation/asciidoc/topics/ref_datagrid_service_resources.adoc (hunk 1 above)
Comment (oraNod) [reply documentation/asciidoc/topics/ref_datagrid_service_resources.adoc:61]: Thanks. I've changed `Infinispan` resources to `{brandname}` resources for downstream.

File: documentation/asciidoc/topics/con_operator_deployments.adoc (hunk 3 above)
Comment (oraNod) [reply documentation/asciidoc/topics/con_operator_deployments.adoc:40]: I've added that as a paragraph to this section. Maybe it's a bit wordy though.

File: documentation/asciidoc/topics/con_operator_deployments.adoc [hunk 4]
```diff
@@ -34,9 +34,14 @@ endif::downstream[]
...
 
-Bidirectional reconciliation synchronizes your CRs with changes that you make to `Infinispan` resources through the {brandname} Console, command line interface (CLI), or other client application and vice versa.
+Bidirectional reconciliation synchronizes your CRs with changes that you make to {brandname} resources through the {brandname} Console, command line interface (CLI), or other client application and vice versa.
 For example if you create a cache through the {brandname} Console then {ispn_operator} adds a declarative Kubernetes representation.
 
-To perform reconciliation {ispn_operator} creates a `listener` pod in each {brandname} cluster that detects modifications for `Infinispan` resources.
+{ispn_operator} associates declarative Kubernetes representations of {brandname} resources with an `Infinispan` CR.
+Deleting an `Infinispan` CR also deletes the {brandname} resources.
+For example deleting an `Infinispan` CR removes any associated `Cache` CRs.
+However deleting an `Infinispan` CR does not remove any {brandname} resources that you create through the console, CLI, or client applications.
```
Comment (ryanemerson) [inline documentation/asciidoc/topics/con_operator_deployments.adoc:45]: This isn't quite right.

Deleting an `Infinispan` CR only removes Cache CRs that were created by the config listen, i.e. not Cache CRs created explicitly by the user.

Deleting an `Infinispan` CR most definitely removes the resources that you create through the console etc as it removes the underlying cluster they are created on :slightly_smiling_face:. It will also remove any Cache CR created by the config listener.

Comment (oraNod) [reply documentation/asciidoc/topics/con_operator_deployments.adoc:45]: :facepalm: I suppose it would help if I actually thought about it.

**Comment by ryanemerson** [comment]
Thanks @oraNod. Can you create a backport PR?
